.github/workflows/validate-files-and-commits.yml

```diff
@@ -3,12 +3,10 @@ name: Validate Files and Commit Messages
 on:
   workflow_call:
     inputs:
-      actions-repo:
+      ref:
```

Comment:
Is the ref here used from the `actions/checkout@v4`? because it does not seem to be used anywhere else.

Reply:
it is an option for further refactoring. All nested/reusable workflows should have this variable to make sure we are testing the intended code. For instance when called from a PR trigger, this variable should follow any nested workflow from `${{ github.event.pull_request.head.sha }}`. If, for instance, we use this variable and the workflow is called from the manual trigger, this entry will be empty. For now it works, as the context where we wake up in the manual trigger is the actual branch. But having this `ref` var is the way to go.